test/io_uring_2_test.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <unifex/config.hpp>
+
+#if !UNIFEX_NO_LIBURING
+
+#  include <unifex/linux/io_uring_context.hpp>
+
+#  include <unifex/finally.hpp>
+#  include <unifex/inplace_stop_token.hpp>
+#  include <unifex/never.hpp>
+#  include <unifex/scheduler_concepts.hpp>
+#  include <unifex/stop_when.hpp>
+#  include <unifex/stream_concepts.hpp>
+#  include <unifex/sync_wait.hpp>
+#  include <unifex/task.hpp>
+#  include <unifex/then.hpp>
+#  include <unifex/when_all.hpp>
+
+#  include <chrono>
+#  include <cstdio>
+#  include <fcntl.h>
+#  include <string>
+#  include <thread>
+
+#  include <gtest/gtest.h>
+#  if !UNIFEX_NO_COROUTINES
+using namespace unifex;
+using namespace unifex::linuxos;
+namespace {
+constexpr std::chrono::milliseconds stopAfter{42};
+const char* fdPath = "/proc/self/fd/";
+
+struct IOUringTest : testing::Test {
+  void SetUp() override {
+    ASSERT_NE(pipe(pipes_), -1) << "unable to create pipe";
+  }
+
+  ~IOUringTest() {
+    close(pipes_[0]);
+    close(pipes_[1]);
```

Comment:
Is this safe to do if the assertion fails in `SetUp`?

Reply:
I am not 100% sure even after reading `man close`. Making this conditional.